x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_text_property.ts

```diff
@@ -9,16 +9,29 @@ import type { Map as MbMap } from '@kbn/mapbox-gl';
 import { DynamicStyleProperty } from './dynamic_style_property';
 import { LabelDynamicOptions } from '../../../../../common/descriptor_types';
 import { RawValue } from '../../../../../common/constants';
+import { IESAggField } from '../../../fields/agg';
 
 export class DynamicTextProperty extends DynamicStyleProperty<LabelDynamicOptions> {
   syncTextFieldWithMb(mbLayerId: string, mbMap: MbMap) {
     if (this._field && this._field.isValid()) {
       const targetName = this.getMbPropertyName();
-      mbMap.setLayoutProperty(mbLayerId, 'text-field', [
-        'coalesce',
-        [this.getMbLookupFunction(), targetName],
-        '',
-      ]);
+
+      if (
+        typeof (this._field as IESAggField).isCount === 'function' &&
+        (this._field as IESAggField).isCount()
+      ) {
+        mbMap.setLayoutProperty(mbLayerId, 'text-field', [
+          'number-format',
+          [this.getMbLookupFunction(), targetName],
+          { locale: navigator.language },
```

Comment:
Should we use the locale here from Kibana, _not_ the browser locale? e.g. it is used for translations and for EMS connections.

cf https://github.com/elastic/kibana/blob/4a988dc5e3875f214564ab03c7f2eb2725cee87e/x-pack/plugins/maps/public/util.ts#L44